core/src/main/java/io/opentelemetry/android/export/BufferedDelegatingExporter.kt

```diff
@@ -0,0 +1,86 @@
+package io.opentelemetry.android.export
+
+import io.opentelemetry.sdk.common.CompletableResultCode
+import java.util.concurrent.atomic.AtomicBoolean
+
+/**
+ * An in-memory buffer delegating span exporter that buffers span data in memory until a delegate is set.
+ * Once a delegate is set, the buffered span data is exported to the delegate.
+ *
+ * The buffer size is set to 5,000 by default. If the buffer is full, the exporter will drop new span data.
+ */
+internal abstract class BufferedDelegatingExporter<T, D>(private val bufferSize: Int = 5_000) {
+    private var delegate: D? = null
+    private val buffer = mutableListOf<T>()
+    private val lock = Any()
+    private var isShutDown = AtomicBoolean(false)
+
+    /**
+     * Sets the delegate for this exporter and flushes the buffer to the delegate.
+     *
+     * If the delegate has already been set, an [IllegalStateException] will be thrown.
+     * If this exporter has been shut down, the delegate will be shut down immediately.
+     *
+     * @param delegate the delegate to set
+     */
+    fun setDelegate(delegate: D) {
+        synchronized(lock) {
+            if (this.delegate != null) {
+                throw IllegalStateException("Exporter delegate has already been set.")
+            }
+
+            this.delegate = delegate
+
+            flushToDelegate(delegate)
```

Comment:
You need to call this before setting the delegate or we might not preserve the order of the logs
```suggestion
            flushToDelegate(delegate)

            this.delegate = delegate
```

Reply:
Ah I see that now, thank you